src/main/java/org/systemexception/simplexdb/database/impl/BerkeleyDbService.java

```diff
@@ -156,13 +157,10 @@ public boolean save(Data data) throws DatabaseException {
 				}
 			}
 			if (usedMemory > maxMemoryOccupation) {
-				logger.warn(LogMessages.MEMORY_OCCUPATION_HIT.toString());
-				foundData.clear();
-				Data warningData = new Data();
-				warningData.setInternalId("WARNING");
-				warningData.setName("Please narrow your search");
-				foundData.add(warningData);
-				return foundData;
+				if (usedMemory > maxMemoryOccupation) {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [These nested if statements could be combined](https://www.codacy.com/app/lcappuccio/simplexdb/file/2273289643/issues/source?bid=3039316&fileBranchId=3185992#l160)